src/libraries/System.Net.WebSockets.Client/tests/SendReceiveTest.Http2.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.Http;
+using System.Net.Sockets;
+using System.Net.Test.Common;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.Net.WebSockets.Client.Tests
+{
+    public abstract class SendReceiveTest_Http2 : ClientWebSocketTestBase
+    {
+        public SendReceiveTest_Http2(ITestOutputHelper output) : base(output) { }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        [SkipOnPlatform(TestPlatforms.Browser, "Self-signed certificates are not supported on browser")]
+        public async Task SendReceive_Success(bool useHandler)
+        {
+            await Http2LoopbackServer.CreateClientAndServerAsync(async uri =>
+            {
+                using (var cws = new ClientWebSocket())
+                using (var cts = new CancellationTokenSource(TimeOutMilliseconds))
+                {
+                    cws.Options.HttpVersion = HttpVersion.Version20;
+                    cws.Options.HttpVersionPolicy = Http.HttpVersionPolicy.RequestVersionExact;
+                    if (useHandler)
+                    {
+                        using var handler = new SocketsHttpHandler();
+                        await cws.ConnectAsync(uri, new HttpMessageInvoker(handler), cts.Token);
+                    }
+                    else
+                    {
+                        await cws.ConnectAsync(uri, cts.Token);
+                    }
+
+                    await cws.SendAsync(new byte[] { 2, 3, 4 }, WebSocketMessageType.Binary, true, cts.Token);
+
+                    await cws.ReceiveAsync(Array.Empty<byte>(), cts.Token);
```

Comment:
I am not sure I understand the meaning of this test. Could you please explain? Both sides are sending at the same time, then the server is not reading reply at all, and client is reading into empty array (meaning also not reading anything the server sent)? I am confused...

Reply:
I'm checking that receive after send on client side does not throw -- it was the original issue. I will add check for received buffer

Reply:
It might be also good to reflect in test name and/or in a comment -- that it's not a general send-receive test, it is testing a specific scenario

Reply:
Fixed